### PR TITLE
fix: directory tree expand when defaultExpandParent is false (#34991)

### PR DESCRIPTION
**Pull request by lalalazero**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

#34991 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     set not to expand at first mount if defaultExpandParent is false      |
| 🇨🇳 Chinese |     设置 defaultExpandParent 为 false 那么首次加载就不展开     |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by lalalazero** [comment]
@MadCcc  看看这样改可以吗

File: components/tree/__tests__/directory.test.js [hunk 1]
```diff
@@ -132,6 +132,17 @@ describe('Directory Tree', () => {
...
 
+  it('should not expand when defaultExpandParent and defaultExpandAll is false', () => {
+    const wrapper = render(
+      createTree({
+        defaultExpandAll: false,
+        defaultExpandParent: false,
+        expandedKeys: ['0-0', '0-1'],
+      }),
+    );
+    expect(wrapper).toMatchSnapshot();
```
Comment (afc163) [inline components/tree/__tests__/directory.test.js:143]: toMatchSnapshot 貌似不能验证用例本身。

Comment (lalalazero) [reply components/tree/__tests__/directory.test.js:214]: 现在可以了不

**Comment by lalalazero** [comment]
@afc163  看看呢

**Comment by afc163** [comment]
可以试试把 defaultExpandParent 和 defaultExpandAll 都传给 rc-tree。

**Comment by lalalazero** [comment]
defaultExpandParent 和 defaultExpandAll 本身就已经传递下去了 https://github.com/ant-design/ant-design/blob/master/components/tree/DirectoryTree.tsx#L205

**Comment by li-jia-nan** [comment]
Hey! Conflicting files, and it seems this is now really outdated. If this is something that still makes sense to the latest master version of Antd, please send a new PR with this change :)

**Pull request closed without merging**